sonar-python-plugin/src/main/java/org/sonar/plugins/python/parser/StaxParser.java

```diff
@@ -0,0 +1,76 @@
+/*
+ * SonarQube Python Plugin
+ * Copyright (C) 2011-2016 SonarSource SA
+ * mailto:contact AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonar.plugins.python.parser;
+
+import com.ctc.wstx.stax.WstxInputFactory;
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import javax.xml.stream.XMLInputFactory;
+import javax.xml.stream.XMLStreamException;
+import org.codehaus.staxmate.SMInputFactory;
+import org.codehaus.staxmate.in.SMHierarchicCursor;
+
+/**
+ * Class copied from deprecated class StaxParser of sonar-plugin-api.
+ */
+public class StaxParser {
+
+  private SMInputFactory inf;
+
+  private XmlStreamHandler streamHandler;
+
+  public StaxParser(XmlStreamHandler streamHandler) {
+    this.streamHandler = streamHandler;
+    WstxInputFactory xmlFactory = (WstxInputFactory) XMLInputFactory.newInstance();
+    xmlFactory.configureForLowMemUsage();
+    xmlFactory.setProperty(XMLInputFactory.IS_VALIDATING, false);
+    xmlFactory.setProperty(XMLInputFactory.SUPPORT_DTD, false);
+    xmlFactory.setProperty(XMLInputFactory.IS_NAMESPACE_AWARE, false);
+    inf = new SMInputFactory(xmlFactory);
+  }
+
+  public void parse(File xmlFile) throws XMLStreamException {
+    try (FileInputStream input = new FileInputStream(xmlFile)) {
+      parse(input);
+    } catch (IOException e) {
+      throw new XMLStreamException(e);
```

Comment:
@pynicolas Theoretically we should, but it would not bring much. In the practical usages, the file always exists (the path is a path to directory, not to a file).